src/visitors/components/SignInForm.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Multi-Stage Sign in Form for Visitors
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { assocPath, evolve, compose } from 'ramda';
+import { Form, PrimaryButton } from '../../shared/components/form/base';
+import LabelledInput from '../../shared/components/form/LabelledInput';
+import LabelledSelect from '../../shared/components/form/LabelledSelect';
+import { BirthYear } from '../../shared/constants';
+import { Visitors } from '../../api';
+import { reduceVisitorsToFields, renameKeys } from '../../util';
+
+
+const CustomForm = styled(Form)`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  width: 100%;
+  padding: 1.2em;
+`;
+
+const Button = styled(PrimaryButton)`
+  width: 100%;
+  padding: 1.2em 0;
+`;
+
+const transformStateToPayload = compose(
+  renameKeys({ birthYear: 'age' }),
+  evolve({ birthYear: y => [y, y].map(BirthYear.toAge) }),
+);
+
+export default class SignInForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      fields: SignInForm.Fields.slice(0, 1),
+      form: {},
+      errors: [],
+      uuid: btoa(new Date().toISOString()), // Used to prevent browser autocomplete
```

Comment:
where is `btoa` coming from

Reply:
It's a browser global: https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/btoa